Dispensers fail to activate() or deactivate()

**SuperTux version:** v0.6.3 [v0.6.2 gives different error, v0.6.1 is unaffected and works as intended] 
**System information:** ---

##### Expected behavior
Given a dispenser named 'disp', the scripts `disp.deactivate();` and `disp.activate();` will stop and start the dispenser.

##### Actual behavior
A 'fatal' error prevents either function from completing:
> v0.6.3
`[FATAL] ./src/scripting/dispenser.cpp:24 error: script is accessing a dead object: 0`
`[FATAL] ./src/scripting/dispenser.cpp:30 error: script is accessing a dead object: 0`

> v0.6.2
`[WARNING] ./src/squirrel/squirrel_environment.cpp:171 Error running script: Squirrel error: Couldn't start script (the index 'disp' does not exist)`

##### Steps to reproduce actual behavior
Example level for testing: [dispenser_script_test.stl.zip](https://github.com/SuperTux/supertux/files/10223468/dispenser_script_test.stl.zip)
1) Jump inside the first script trigger to run `disp.deactivate();`, an error will appear in the console. The dispenser does not deactivate as intended.
2) Jump inside the second script trigger to run `disp.activate();`, an error will appear in the console

--- a/src/scripting/badguy.hpp
+++ b/src/scripting/badguy.hpp
@@ -34,7 +34,9 @@ class BadGuy
 {
 #ifndef SCRIPTING_API
 public:
-  using GameObject::GameObject;
+  BadGuy(UID uid) :
+    GameObject<::BadGuy>(uid)
+  {}
 
 private:
   BadGuy(const BadGuy&) = delete;

--- a/src/scripting/dispenser.hpp
+++ b/src/scripting/dispenser.hpp
@@ -34,8 +34,8 @@ class Dispenser final : public scripting::BadGuy
 #ifndef SCRIPTING_API
 public:
   Dispenser(UID uid) :
-    BadGuy(uid),
     GameObject<::BadGuy>(uid),
+    BadGuy(uid),
     GameObject<::Dispenser>(uid)
   {}
 

--- a/src/scripting/dispenser.hpp
+++ b/src/scripting/dispenser.hpp
@@ -18,8 +18,8 @@
 #define HEADER_SUPERTUX_SCRIPTING_DISPENSER_HPP
 
 #ifndef SCRIPTING_API
+
 #include "scripting/badguy.hpp"
-#include "scripting/game_object.hpp"
 
 class Dispenser;
 #endif
@@ -33,7 +33,11 @@ class Dispenser final : public scripting::BadGuy
 {
 #ifndef SCRIPTING_API
 public:
-  using BadGuy::BadGuy;
+  Dispenser(UID uid) :
+    BadGuy(uid),
+    GameObject<::BadGuy>(uid),
+    GameObject<::Dispenser>(uid)
+  {}
 
 private:
   Dispenser(const Dispenser&) = delete;

--- a/src/scripting/dispenser.hpp
+++ b/src/scripting/dispenser.hpp
@@ -35,8 +35,8 @@ class Dispenser final : public scripting::BadGuy
 public:
   Dispenser(UID uid) :
     GameObject<::BadGuy>(uid),
-    BadGuy(uid),
-    GameObject<::Dispenser>(uid)
+    GameObject<::Dispenser>(uid),
+    BadGuy(uid)
   {}
 
 private:

--- a/src/scripting/game_object.hpp
+++ b/src/scripting/game_object.hpp
@@ -87,10 +87,6 @@ template<class T>
 class GameObject
 {
 public:
-  GameObject() :
-    m_uid()
-  {}
-
   GameObject(UID uid) :
     m_uid(uid)
   {}

--- a/src/scripting/willowisp.hpp
+++ b/src/scripting/willowisp.hpp
@@ -36,8 +36,8 @@ class WillOWisp final : public scripting::BadGuy
 public:
   WillOWisp(UID uid) :
     GameObject<::BadGuy>(uid),
-    BadGuy(uid),
-    GameObject<::WillOWisp>(uid)
+    GameObject<::WillOWisp>(uid),
+    BadGuy(uid)
   {}
 
 private:

--- a/src/scripting/willowisp.hpp
+++ b/src/scripting/willowisp.hpp
@@ -35,8 +35,8 @@ class WillOWisp final : public scripting::BadGuy
 #ifndef SCRIPTING_API
 public:
   WillOWisp(UID uid) :
-    BadGuy(uid),
     GameObject<::BadGuy>(uid),
+    BadGuy(uid),
     GameObject<::WillOWisp>(uid)
   {}
 

--- a/src/scripting/willowisp.hpp
+++ b/src/scripting/willowisp.hpp
@@ -19,8 +19,8 @@
 
 #ifndef SCRIPTING_API
 #include <string>
+
 #include "scripting/badguy.hpp"
-#include "scripting/game_object.hpp"
 
 class WillOWisp;
 #endif
@@ -34,7 +34,11 @@ class WillOWisp final : public scripting::BadGuy
 {
 #ifndef SCRIPTING_API
 public:
-  using BadGuy::BadGuy;
+  WillOWisp(UID uid) :
+    BadGuy(uid),
+    GameObject<::BadGuy>(uid),
+    GameObject<::WillOWisp>(uid)
+  {}
 
 private:
   WillOWisp(const WillOWisp&) = delete;

